ls1intum/Artemis
# Feature/programming exercises/migrate remarkable to showdown

### Checklist

### Motivation and Context



While the markdown editor uses [showdown JS](https://github.com/showdownjs/showdown), the programming exercise instructions use [remarkable](https://github.com/jonschlinkert/remarkable).
We want to only have 1 dependency in our project for markdown parsing, so we are migrating the programming exercise instructions to showdown.

### Description


The current implementation of interactive instruction parsing was transferred from remarkable to showdown, using the [showdown extension system](https://github.com/showdownjs/showdown/wiki/extensions):
- Parsing of tasks (including test cases)
- Parsing of plantUmls (including validation of successful tests)

Improvements made in this PR:
- In the markdown editor the instructions are only parsed when the preview tab is opened (before it happened on every editor change callback)
- The tasks are now their own component
- The step wizard is now a component
- Katex is now available in the interactive instructions

How the extensions work:
- They parse the given element from the string
- The replace the found instances of the given element with a div container
- They provide the component a callback function that inserts the Components with the Angular Injector once the rendered html is rendered in the Angular template

### Steps for Testing


1. Log in to ArTEMiS
2. Go to student code-editor
3. Do a submission and make sure the programming exercise instructions reflect the result

1. Log in to ArTEMiS
2. Go to programming-exercises > edit
3. Update the markdown and check if it rendered correctly

### Screenshots



![image](https://user-images.githubusercontent.com/28230611/62005041-a0be9500-b12d-11e9-8dad-4a91fd19e744.png)
